### PR TITLE
Add `jetstream::Stream::get_raw_message`

### PR DESCRIPTION
**Pull request by caspervonb**

### CODE REVIEW

File: async-nats/src/jetstream/stream.rs [hunk 1]
```diff
@@ -331,6 +350,36 @@ pub struct StreamState {
     pub consumer_count: usize,
 }
 
+/// A raw stream message in the representation it is stored.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct Message {
```
Comment (caspervonb) [inline async-nats/src/jetstream/stream.rs:355]: Alternative name:

```suggestion
pub struct RawMessage {
```

We'll end up having `DirectMessage` too as that was just added to the server.

File: async-nats/src/jetstream/stream.rs [hunk 2]
```diff
@@ -331,6 +350,36 @@ pub struct StreamState {
...
+    pub headers: Option<String>,
+
+    /// The time the message was published.
+    #[serde(rename = "time", with = "rfc3339")]
+    pub time: time::OffsetDateTime,
+}
+
+/// Helper type to deserialize and flatten message requests
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct MessageStatus {
```
Comment (caspervonb) [inline async-nats/src/jetstream/stream.rs:379]: Ideally, probably flatten this

```suggestion
pub struct MessageStatus {
```

File: async-nats/src/jetstream/stream.rs [hunk 3]
```diff
@@ -33,6 +33,25 @@ pub struct Stream {
...
+        let subject = format!("{}.STREAM.MSG.GET.{}", self.prefix, &self.info.config.name);
+        let payload = json!({
+            "seq": sequence,
+        });
+
+        match self.context.request(subject, &payload).await? {
+            Response::Err { error } => Err(Box::new(std::io::Error::new(
+                ErrorKind::Other,
+                format!(
+                    "nats: error while creating stream: {}, {}",
```
Comment (caspervonb) [inline async-nats/src/jetstream/stream.rs:47]:
```suggestion
                    "nats: error while getting message: {}, {}",
```

File: async-nats/src/jetstream/stream.rs [hunk 4]
```diff
@@ -32,6 +32,26 @@ pub struct Stream {
 }
 
 impl Stream {
+    /// Get a raw message from the stream.
```
Comment (Jarema) [inline async-nats/src/jetstream/stream.rs:35]: Please add docs.

File: async-nats/src/jetstream/stream.rs [hunk 5]
```diff
@@ -321,6 +341,35 @@ pub struct StreamState {
...
+    #[serde(rename = "subject")]
+    pub subject: String,
+
+    /// Sequence of the message.
+    #[serde(rename = "seq")]
+    pub sequence: u64,
+
+    /// Data of the mssage.
+    #[serde(default, rename = "data")]
+    pub data: String,
```
Comment (Jarema) [inline async-nats/src/jetstream/stream.rs:357]: why not bytes?

**Review by Jarema — CHANGES_REQUESTED** [review]

File: async-nats/src/jetstream/stream.rs (hunk 5 above)
Comment (caspervonb) [reply async-nats/src/jetstream/stream.rs:357]: Yeah base64 encoded string, with terrible documentation from legacy nats.rs (from me.. gotta admit) 😁

Comment (Jarema) [reply async-nats/src/jetstream/stream.rs]: ah, ok.

Yes, `TryInto` is a nice pattern from those fallible transofmartions we have in library.

**Review by Jarema — APPROVED** [review]
LGTM

**Pull request merged**